fdbclient/NativeAPI.actor.cpp

```diff
@@ -2100,6 +2100,7 @@ ACTOR Future<Void> watch( Reference<Watch> watch, Database cx, Transaction *self
 	cx->addWatch();
 	try {
 		self->watches.push_back(watch);
+		self = nullptr; // This actor may outlive *self
```

Comment:
Since, `self` is an argument to `watch`, and the contract is actor can outlive `self`, perhaps this comment should go to the function documentation, so that contract is immediately visible to user of this actors.

Reply:
We could potentially also move the starting logic out of the actor and avoid passing in `self` altogether. This would be done by moving

```cx->addWatch()
self->watches.push_back(watch)```

to the calling function and passing `info` in as a parameter.

Reply:
Hi @atn34 , are you planning to change the code as AJ suggested and have this PR merged after that?

Reply:
Done, sorry for sitting on this for so long

Reply:
Thanks! I will kick off a correctness test before merge it.